SecretSanta/src/SecretSanta.Api/Controllers/UsersController.cs

```diff
@@ -43,17 +71,30 @@ public ActionResult Delete(int id)
         }
 
         [HttpPost]
-        public ActionResult<User?> Post([FromBody] User? user)
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(FullUser), StatusCodes.Status200OK)]
+        //public ActionResult<User?> Post([FromBody] User? user)
+        public ActionResult<FullUser?> Post([FromBody] FullUser? user)
         {
             if (user is null)
             {
                 return BadRequest();
             }
-            return Repository.Create(user);
+            //return Repository.Create(user);
+            Repository.Create(new User{
+                FirstName = user.FirstName,
+                LastName = user.LastName,
+                Id = (Repository.List().Select(item => item.Id).Max() + 1)
+            });
+
+            return user;
         }
 
         [HttpPut("{id}")]
-        public ActionResult Put(int id, [FromBody] User? user)
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public ActionResult Put(int id, [FromBody] FullUser? user)
```

Comment:
In this case it looks like you forgot to change the parameter to `UpdateUser`
```suggestion
        public ActionResult Put(int id, [FromBody] UpdateUser? user)
```